src/itty-fetcher.ts

```diff
@@ -39,14 +39,24 @@ const fetchy = (options: FetchyOptions): FetchyFunction => (
   payload?: string | number | object | undefined,
   fetchOptions?: FetchOptions,
 ) => {
-  const {
-    base,
-    autoParse,
-    method,
-  } = options
+  const method = options.method.toUpperCase()
+  const resolvedURL = new URL(options.base + url)
+
+  /**
+   * If the request is a `.get(...)` then we want to pass the payload
+   * to the URL as query params as passing data in the body is not
+   * allowed for GET requests. We clear the payload after this so that
+   * it doesn't get passed to the body.
+   */
+  if (method === 'GET' && payload && typeof payload === 'object') {
+    resolvedURL.search = new URLSearchParams(
+      payload as Record<string, string>
```

Comment:
This should probably allow for passing URLSearchParams directly, as is you can't pass multilpes (a frequent strategy for 'arrays': `?hello=world&hello=darkness&hello=cześć` for `hello: ["world", "darkness", "cześć" ]`

Reply:
You're right, right now it would result in `hello=world%2Cdarkness%2Ccześć`. Will look into it 👍

Reply:
Ok, pushed a change to support either/or 😄